v_frame/src/plane.rs

```diff
@@ -138,6 +140,37 @@ impl<T: Pixel> std::ops::Drop for PlaneData<T> {
   }
 }
 
+#[cfg(feature = "serialize")]
+impl<T: Pixel + Serialize> Serialize for PlaneData<T> {
+  fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+  where
+    S: serde::Serializer,
+  {
+    use serde::ser::SerializeSeq;
+
+    let mut data = serializer.serialize_seq(Some(self.len))?;
+    for byte in
+      unsafe { std::slice::from_raw_parts(self.ptr.as_ptr(), self.len) }
```

Comment:
plane data derefs to &[] you can avoid unsafe here

Reply:
I tried to do a deref but couldn't figure out the magic syntax to make the compiler happy. I guess I could always do `.deref()`